#!/usr/bin/env python
PACKAGE = "four_way_stop"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add('initiate', bool_t, 0, 'Start', False)
gen.add('speed', double_t, .1, 'Speed', 2.5, 0, 5)
gen.add('maxTurn', double_t, .1, 'Max Turn Angle', 1.3, 0, 1.5)
gen.add('turningAgnle', double_t, .05, 'Max Turn Angle', .1, 0, 1)
gen.add('waypointFoundDistance', double_t, .000005, 'Waypoint Found Distance', .00003, .000005, .0001)

gen.add('maxRange', int_t, .1, 'Max Range', 20,  0,    50)
gen.add('minRange', int_t, .1, 'Min Range', 2,   0,    10)
gen.add('maxIndex', int_t,  1, 'Max Index', 10,  -180, 180)
gen.add('minIndex', int_t,  1, 'Min Index', -30, -180, 180)


exit(gen.generate(PACKAGE, "four_way_stop", "commander"))